### ISSUE TITLE
Bug or just mee !? "Re-Color Mesh"

### ISSUE THREAD

**Issue by arkdanielnielsen** [mostaphaRoudsari/ladybug#131]
1) Something goes wrong when I want to re-color a result from "Color Zones by EP Result". 
2) I am not sure in which component things goes wrong. It seems to be something with the meshing? Or is it the "Re-Color Mesh"?
3) As seen in the image it is 8 identical simple shoebox'es. I assume they should have the same mesh size, but that is not the case. Original the geometry derived from a Rulled Surface from two straight lines.
4) Why I am using is just to remove the legend text or want is called Analysis Text in the re-color mesh component. Would it be possible to have the Analysis text as a out in the different original components like the "Color Zones by EP Result" etc? That would in many cases be quite useful - at least what I think ;)
![ladybug_re-color-mesh-01](https://cloud.githubusercontent.com/assets/7299731/6135261/768953da-b167-11e4-96cc-8acaecaf046f.jpg)

**Comment by chriswmackey** [comment]
@arkdanielnielsen ,
1) Can't you just use the legendPar of the "Color Zones by EP Result" component?  Are you saying that is not working?
2) Just don't graft the list of meshes coming out of the colorZones component and then the lists should probably align.  Alternatively, you can graft the analysis result.
3) The auto-meshing algorithms of GH might divide something up into quads or triangles.  It is a difficult process to control unless you make the mesh yourself from scratch.
4) Ok.  I understand the answer to my fist question.  I do not understand what you mean by Analysis Text.  You you mean have title and legend text separately?

**Comment by chriswmackey** [comment]
Ah wait.  I just understood why the reColorMesh does not work.  Here's a workaround but you probably don't want to use the recolor mesh component in this case.
![image](https://cloud.githubusercontent.com/assets/5567574/6156039/a5f1ae18-b204-11e4-95b4-cb5522e0242f.png)

Would having separate outputs for title and legend text fix this issue for you?  If so, that seems like the best route.
-Chris

**Comment by arkdanielnielsen** [comment]
The thing is, I want to get rid of the Analysis Title (it was a mistake I wrote Analysis text) and keep the legend. To do that I have to cull away the text but not in all cases the length of the analysis title is the same, so that makes it difficult to use that method. This is not only a wish for "Color Zones by EP Result" but in general for all components with legends. I think it would be easier to just remove or place the Analysis Title without using the reColorMesh.

**Comment by chriswmackey** [comment]
@arkdanielnielsen ,
I agree with you that it would be nice to have the outputs separate and I just made the change for the color zones and color surfaces component.
![image](https://cloud.githubusercontent.com/assets/5567574/6179290/c34b7250-b2e5-11e4-9798-bd85392b5e5b.png)

If you draw up a list of components where this grouping of legend and title outputs happens, I will begin working my way down it (it looks like this issue will take a lot longer to close than initially anticipated :).  Feel free to order the list in terms of the components that you would most like to see the change and I will get to those first.

-Chris

**Comment by arkdanielnielsen** [comment]
Looks good, thanks.
It sound like a deal, that I make the list. Maybe I could add the lines if it is the same lines there is needed in the different components?

**Comment by arkdanielnielsen** [comment]
@chriswmackey finally I had time. In the gh file I have marked all component where the output of AnalysisTitle could be useful.
https://www.dropbox.com/s/3kni0fqm0ozt7w0/all_components_LBHB.gh?dl=0
Is it only to add the below lines? If so I could try to add them in the components.

    analysisTitle = []
    for count, item in enumerate(legendInit):
        if count == 0:
            legend.append(item)
        if count == 1:
            for srf in item:
                legend.append(srf)
        if count == 2:
            for srf in item:
                analysisTitle.append(srf)

**Comment by mostaphaRoudsari** [comment]
@chriswmackey @arkdanielnielsen what is up with this issue? should we close this?

**Comment by arkdanielnielsen** [comment]
@mostaphaRoudsari It is about separating the Analysis Text from the legend output in two separated outputs. To make it easier to remove the text of legend if it is not wanted. And I think it could be useful for all the components with legends.

**Comment by chriswmackey** [comment]
@arkdanielnielsen and @mostaphaRoudsari ,

I have changed the name of the issue to be closer to what we have decided to do.  In my view, it is a long-term issue as I have a lot of other priorities on my list and there are a bunch of components to change.  All that has to happen is that the code that @arkdanielnielsen posted has to go into a lot of the components and a new output has to be added to them.  @arkdanielnielsen , if you want to take this up on your own, that would be swell.

Let me know your anticipated time frame and if you run into any issues.
-Chris

**Comment by chriswmackey** [comment]
I just realized that I cannot assign the issue to @arkdanielnielsen .  I will put myself back as asignee.

**Comment by chriswmackey** [comment]
I have gotten this separation of title and legend working now for a large fraction of the Ladybug components.  I feel that I should just finish this off for the next stable release.

**Comment by chriswmackey** [comment]
This feature will be addressed with the new core Ladybug functions that @mostaphaRoudsari is putting together and so I will let this issue rest until this core development is finished.

**Comment by chriswmackey** [comment]
This is a monumental task that we will fix in LB+ but I welcome anyone who wants to implement this on any of the legacy components.
